zebra-chain/src/transaction/hash.rs

```diff
@@ -19,11 +19,10 @@ pub struct Hash(pub [u8; 32]);
 
 impl<'a> From<&'a Transaction> for Hash {
     fn from(transaction: &'a Transaction) -> Self {
-        let mut hash_writer = sha256d::Writer::default();
-        transaction
-            .zcash_serialize(&mut hash_writer)
-            .expect("Transactions must serialize into the hash.");
-        Self(hash_writer.finish())
+        let hasher = TxIdHasher::new(&transaction);
+        hasher
+            .txid()
+            .expect("zcash_primitives and Zebra transaction formats must be compatible")
     }
 }
```

Comment:
Is this rule held with ZIP-244 transaction id's here?

https://doc.rust-lang.org/std/hash/trait.Hash.html#hash-and-eq
![image](https://user-images.githubusercontent.com/552961/124224897-82f5c980-dad4-11eb-873f-8c5c47495405.png)

Reply:
The rule doesn't apply, because `Hash` is our custom `transaction::Hash` struct, not the Rust `std::hash::Hash` trait.

(A similar rule applies due to the Zcash consensus rules around transaction IDs.)

If we want to rename the struct or `Transaction::hash` method, let's open a cleanup ticket.